docker/Dockerfile.linux

```diff
@@ -1,21 +1,43 @@
-FROM mono:6.12
-
-MAINTAINER Justin Phelps <linuturk@onitato.com>
+ARG monoversion="6.12"
+FROM mono:${monoversion} as build
 
 RUN apt-get update && apt-get install git -y
 
 COPY . /usr/local/src/choco/
 
 WORKDIR /usr/local/src/choco
-RUN chmod +x build.sh
-RUN chmod +x zip.sh
-RUN ./build.sh
+RUN chmod +x *.sh
+
+ARG buildscript="build.sh"
+RUN ./$buildscript
+
+RUN if [ "$buildscript" = "build.official.sh" ]; then \
+    cp docker/choco_official_wrapper code_drop/chocolatey/console/choco_wrapper; \
+  else \
+    cp docker/choco_wrapper code_drop/chocolatey/console/choco_wrapper; \    
+  fi;
+
+
+ARG monoversion="6.12"
```

Comment:
Does this need to be duplicated?

Reply:
Yes, as the variable goes out of scope for every build stage (i.e. each `FROM), as per:
https://docs.docker.com/engine/reference/builder/#scope

Reply:
Gotcha, makes sense. Thanks for confirming.